packages/node/src/utils/substrate.ts

```diff
@@ -247,8 +250,12 @@ export async function fetchBlocksRange(
 ): Promise<SignedBlock[]> {
   return Promise.all(
     range(startHeight, endHeight + 1).map(async (height) => {
-      const blockHash = await api.rpc.chain.getBlockHash(height);
-      return api.rpc.chain.getBlock(blockHash);
+      try {
+        const blockHash = await api.rpc.chain.getBlockHash(height);
+        return api.rpc.chain.getBlock(blockHash);
+      } catch (err) {
+        logger.error(err, `failed to fetch block at height ${height}`);
```

Comment:
I don't know the logger behaviour, but should the error be rethrown or does the logger rethrow?

Reply:
Thanks, you are right, should be rethrown rather than just log it.